libevmasm/EVMAssemblyStack.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libevmasm/EVMAssemblyStack.h>
+
+#include <libsolutil/JSON.h>
+#include <liblangutil/Exceptions.h>
+#include <libsolidity/codegen/CompilerContext.h>
+
+using namespace solidity::util;
+using namespace solidity::langutil;
+using namespace solidity::frontend;
+using namespace std;
+
+namespace solidity::evmasm
+{
+
+void EVMAssemblyStack::parseAndAnalyze(string const& _sourceName, string const& _source)
+{
+	solAssert(!m_evmAssembly);
+	m_name = _sourceName;
+	solRequire(jsonParseStrict(_source, m_json), AssemblyImportException, "Could not parse JSON file.");
+	auto result = evmasm::Assembly::fromJSON(m_json);
```

Comment:
```suggestion
	auto result = Assembly::fromJSON(m_json);
```
We don't need the `evmasm` qualifier, it could be removed here and elsewhere in this file.